docs/TranslationGuide.md

```diff
@@ -1,49 +1,175 @@
 ```
-ZONEMASTER_DISTRIBUTION/zonemaster-gui/src/app/components/navigation.component.html
+1. [What is Zonemaster?](#q1)
+
+#### 1. What is Zonemaster? <a name="q1"></a>
 ```
 
+## Add the language to HTML code
+
+The language must be linked from the GUI. The new language must be added
+to [src/app/components/navigation/navigation.component.html]:
+
 In `navigation.component.html` :
 
- - Locate the div `<div class="lang">`
- - Add your language using the language tag: `<a lang='xx' (click)="setLanguage('xx')">XX</a> |`
-with xx, the language code in lower case and XX, the displayed language code.
- - Finally, for small (mobile) screen, add in the next select element:
- ``` 
+  * Locate the div `<div class="lang">`
+  * Add the language where `xx` is the language code in lower case, e.g.
+    `en` and `Yy` is the same language code with first letter in upper
+    case, e.g. `En`:
+
+```
+<a lang='xx' (click)="setLanguage('xx')">Yy</a> |
+```
+  * Add the language a second time where `xx` is the language code in
+    lower case, e.g. `sv`, and `Yyyyy` is the name of the same language
+    in that language, e.g. `Svenska`:
+
+```
  <option lang='xx' value='xx' [selected]="lang === 'xx'">
-  Xxxxx
+  Yyyyy
  </option>
  ```
- with Xxxxx, the full language name (En => English).
- 
- 
-4. Optionally, to change the default language update the following file: 
+
+Preserve the alphabetical order of the language codes.
+
+## Add the language in Type Script code
+
+The new language must be added to the following files:
+
+* [package.json]
+* [src/app/app.module.ts]
+* [src/app/components/navigation/navigation.component.ts]
+
+### package.json
+
+In `package.json` locate
+
 ```
-ZONEMASTER_DISTRIBUTION/zonemaster-gui/src/app/components/navigation.component.ts
+    "i18n:extract": "ngx-translate-extract --input ./src --output ./src/assets/i18n/{en,da,fr,sv}.json --key-as-default-value --clean --sort --format json"
 ```
-and replace `public lang = 'en'` by `public lang = 'xx';` with xx the language code of your language.
+and add the two-letter language code of the new language. Preserve
+the alphabetical order of the language codes.
+
+### app.module.ts
+
+In `app.module.ts` locate
+
+```
+import 'moment/locale/fr';
+import 'moment/locale/sv';
+```
+and add a new line with code of the new language instead of `xx`
+```
+import 'moment/locale/xx';
+```
+Preserve the alphabetical order of the language codes.
+
+### navigation.component.ts
+
+In `navigation.component.ts` locate
+
+```
+  private isValidLanguage(lang: string) {
+    const validLanguages = [ 'da', 'en', 'fr', 'sv' ];
+```
+and add the two-letter language code of the new language. Preserve
+the alphabetical order of the language codes.
+
+
+## Change default language
+
+Optionally, to change the default language update
+[src/app/components/navigation/navigation.component.ts].
+
+In `navigation.component.ts` update the following code
+```
+  public lang = 'en';
+  private lang_default = 'en';
```

Comment:
From my understanding of the code, only `lang_default` need to be updated.

Reply:
That update will rarely be done, but say that `en` is removed and `da` is the default language, then I guess both must be update. I'll keep it as is, and let us try to find out what is best. Also see #170.